rust/frontend/src/tower_tracing.rs

```diff
@@ -0,0 +1,54 @@
+use axum::extract::MatchedPath;
+use axum::http::{header, Request, Response};
+use axum::Router;
+use std::time::Duration;
+use tower_http::trace::{MakeSpan, OnResponse, TraceLayer};
+
+#[derive(Clone)]
+struct RequestTracing;
+impl<B> MakeSpan<B> for RequestTracing {
+    fn make_span(&mut self, request: &Request<B>) -> tracing::Span {
+        let http_route = request
+            .extensions()
+            .get::<MatchedPath>()
+            .map_or_else(|| "", |mp| mp.as_str())
+            .to_owned();
+
+        let host = request
+            .headers()
+            .get(header::HOST)
+            .map_or("", |h| h.to_str().unwrap_or(""));
+
+        let name = format!("{} {}", request.method(), http_route);
+
+        tracing::span!(
+            tracing::Level::DEBUG,
+            "HTTP request",
+            http.method = %request.method(),
+            http.uri = %request.uri(),
+            http.route = http_route,
+            http.version = ?request.version(),
+            http.host = %host,
+            http.status_code = tracing::field::Empty,
```

Comment:
this was very frustrating
apparently you cannot do `span.record("foo", ...)` unless you created the span with `foo = Empty`